### PR TITLE
Friction Energy Output in H3D file

### PR DESCRIPTION
**Pull request by MahaHachani**
#### Checklist

- [x] The title of the PR is reviewed
- [x] There is no text before the checklist section
- [x] The following two sections are filled (or deleted)
- [x] This PR has been previewed 

Description of the feature or the bug

New development for outputting friction energie in H3d file
/H3D/NODA/CSE_FRICG
/H3D/NODA/CE_FRIC/INTER=ID or ALL
Description of the changes

New development for outputting friction energie in H3d file:
The energy is stored in variables EFRIC and EFRICG
Communication spmd for remote nodes
Before outputting the energy is divided by the surface

### CODE REVIEW

File: engine/source/mpi/interfaces/spmd_exch_efric.F [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+#include      "spmd_c.inc"
+#include      "units_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER  , INTENT(IN) ::  
+     .        NBINTC,ISLEN7, IRLEN7,IRLEN7T, ISLEN7T,       
+     .        IRLEN20, ISLEN20, IRLEN20T, ISLEN20T,
+     .        IPARI(NPARI,NINTER), INTLIST(NBINTC)
+      TYPE(INTBUF_STRUCT_),INTENT(IN) ::  INTBUF_TAB(*)
```
Comment (sebastienVilleneuve) [inline engine/source/mpi/interfaces/spmd_exch_efric.F:80]: @MahaHachani 

you need to put size of the array

File: engine/source/mpi/interfaces/spmd_exch_sorting_efric.F [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+#include      "units_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER  , INTENT(IN) ::  
+     .        NBINTC,ISLEN7, IRLEN7,IRLEN7T, ISLEN7T,       
+     .        IRLEN20, ISLEN20, IRLEN20T, ISLEN20T,
+     .        IPARI(NPARI,NINTER), INTLIST(NBINTC)
+
+      TYPE(INTBUF_STRUCT_),INTENT(IN) ::  INTBUF_TAB(*)
```
Comment (sebastienVilleneuve) [inline engine/source/mpi/interfaces/spmd_exch_sorting_efric.F:81]: size of the array must be specified

File: engine/source/output/h3d/spmd/spmd_h3d_sum_r_nodal_value.F [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+C   C o m m o n   B l o c k s
+C-----------------------------------------------
+#include      "scr05_c.inc"
+#include      "task_c.inc"
+#include      "spmd_c.inc"
+#include      "com01_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER  , INTENT(IN) :: LEN,LENP0, NODGLOB(*)
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/spmd/spmd_h3d_sum_r_nodal_value.F:48]: size of the array must be specified

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]
@MahaHachani 

some other issues to correct

File: engine/source/output/h3d/spmd/spmd_h3d_sum_r_nodal_value.F (hunk 3 above)
Comment (MahaHachani) [reply engine/source/output/h3d/spmd/spmd_h3d_sum_r_nodal_value.F]: Done Sebastien
Thanks

File: engine/source/input/lectur.F [hunk 4]
```diff
@@ -1817,7 +1819,52 @@ SUBROUTINE LECTUR(ICODE   ,ISKEW ,ISKWN  ,IXTG   ,IXS   ,
...
+         NINEFRIC = 0
+         IF(NINTSTAMP/=0) THEN
+           DO N=1,NINTER
+               NI = H3D_DATA%N_CSE_FRIC_INTER (N) 
+               IF(NI/= 0.AND.IPARI(7,N)==21) NINEFRIC_STAMP = NINEFRIC + 1
+           ENDDO
+         ENDIF
+         IF(NINEFRIC_STAMP==NINEFRICG) THEN
+            ALLOCATE (EFRIC_STAMP(NINEFRICG,NUMNODG))
+            EFRIC_STAMP(1:NINEFRICG,1:NUMNODG) = ZERO
```
Comment (sebastienVilleneuve) [inline engine/source/input/lectur.F:1838]: @MahaHachani 

please use MY_ALLOCATE routines

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]

File: engine/source/output/h3d/spmd/spmd_h3d_sum_r_nodal_value.F (hunk 3 above)
Comment (MahaHachani) [reply engine/source/output/h3d/spmd/spmd_h3d_sum_r_nodal_value.F]: Done

**Review by sebastienVilleneuve — APPROVED** [review]

**Pull request merged**